refactor(Commandhandling): rewrite command handling pages to be based upon interactions (#683)
discordjs/guide: guide/creating-your-bot/adding-more-commands.md

:::

A bot with nothing but a single command would be boring, and you probably have a bunch of command ideas floating around in your head already, right? Let's begin, then.

Here's what your message event should currently look like:
Here's what your interaction event should currently look like:

```js
client.on('message', message => {
	if (message.content === '!ping') {
		message.channel.send('Pong.');
client.on('interactionCreate', async interaction => {
	if (!interaction.isCommand()) return;

	if (interacton.commandName === 'ping') {
		await interaction.reply('Pong!');
	}
});
```

Before doing anything else, make a property to store the prefix you've configured. Instead of `const config = ...`, you can destructure the config file to extract the prefix and token variables.
Before doing anything else, make a property to store the token. Instead of `const config = ...`, you can destructure the config file to extract the token variable.

```js {1,3}

> const { prefix, token } = require('./config.json');

Reviewer: Line 29 below could be replaced with the following (maybe also make it a tip?).

```
From now on, if you change the token in your `config.json` file, it'll change in your bot file as well!
```

You already have an if statement that checks messages for a ping/pong command. Adding other command checks is just as easy; chain an `else if` to your existing condition.

```js {2-8}
client.on('interaction', async interaction => {
	if (!interaction.isCommand()) return;

	if (interacton.commandName === 'ping') {
		await interaction.reply('Pong!');
	} else if (interacton.commandName === 'beep') {
		await interaction.reply('Boop!');
	}
});
```

> 

Reviewer 2: ```suggestion

```
obsolete additional newline

You already have an if statement that checks messages for a ping/pong command. Adding other command checks is just as easy; chain an `else if` to your existing condition.

```js {2-8}
client.on('interactionCreate', async interaction => {
	if (!interaction.isCommand()) return;

	if (interacton.commandName === 'ping') {
		await interaction.reply('Pong!');
	} else if (interacton.commandName === 'beep') {
		await interaction.reply('Boop!');
	}
});
```

## Displaying real data

Let's start displaying some real data. For now, we'll be displaying basic member/server info.

### Server info command

> Make another if statement to check for commands using `server` as the command name. You get the interaction object, reply to the interaction just as before:

Reviewer 3: ```suggestion
Make another if statement to check for commands using `server` as the command name. You get the interaction object and reply to the interaction just as before:
```

<!-- eslint-skip -->

```js {10-12}
client.on('interactionCreate', async interaction => {
	if (!interaction.isCommand()) return;

	if (interacton.commandName === 'ping') {
		await interaction.reply('Pong!');
	} else if (interacton.commandName === 'beep') {
		await interaction.reply('Boop!');
	} else if (interacton.commandName === 'server') {
		await interaction.reply(`This server's name is: ${interaction.guild.name}`);
	} else if (interacton.commandName === 'user-info') {
		await interaction.reply(`Your username: ${interaction.user.username}\nYour ID: ${interaction.user.id}`);
	}
});
```

This will display the command author's **username** (not nickname, if they have one set), as well as their user ID.

<DiscordMessages>
	<DiscordMessage profile="bot">
		<template #interactions>
			<DiscordInteraction profile="user" :command="true">user-info</DiscordInteraction>
		</template>
		Your username: User <br>

> 		Your ID: 20833034795932416

Reviewer 3: ```suggestion
		Your ID: 123456789012345678
```
Old ID is probably one of my deleted applications

	</DiscordMessage>
</DiscordMessages>

::: tip

> `interaction.user` refers to the user who sent the command. For a full list of all the properties and methods for the user object, check out <docs-link path="class/User">the documentation page for it</docs-link>.

Reviewer 3: ```suggestion
`interaction.user` refers to the user who sent the command. For a full list of all the properties and methods for the user object, check out <DocsLink path="class/User">the documentation page for it</DocsLink>.
```
